### ISSUE TITLE
Comparator definition does not imply total ordering

### ISSUE THREAD

**Issue by kuprel** [dart-lang/sdk#35354]
In the comparable documentation (https://github.com/dart-lang/sdk/blob/master/sdk/lib/core/comparable.dart) it says:

> A comparison function represents an ordering on a type of objects. A total ordering on a type means that for two values, either they are equal or one is greater than the other (and the latter must then be smaller than the former).
> 
> A Comparator function represents such a total ordering by returning
> 
> - a negative integer if a is smaller than b,
> - zero if a is equal to b, and
> - a positive integer if a is greater than b.

This is insufficient to define a total ordering since transitivity must also be satisfied (a > b and b > c implies a > c).

**Comment by lrhn** [comment]
Correct. We do say that it's an "ordering", but do not say what that means (transitive, anti-symmetric, usually anti-reflexive). 
It's then a total ordering if it's an ordering and all pairs of elements are related.
